src/backends/tensorrt/plan_backend.cc

```diff
@@ -2702,9 +2780,11 @@ PlanBackend::Context::Run(
     }
   }
 
-  // Ensure inputs are ready before execution. Output buffers will always be
-  // available at this point as the execution and output copy are on the same
-  // stream.
+  // Ensure inputs are ready before execution.
+  // Output buffers are guaranteed to be available at this point when the
+  // execution and output copy are on the same stream. If a separate stream is
+  // being used to copy the output, the user provides the guarantee that the
```

Comment:
We still can't have "user provides guarantee". This code must make sure that there is no race condition where buffers could be incorrectly used.

Reply:
Yep, I forgot to update that from the previous commit. Updated.